pandas/core/window/ewm.py

```diff
@@ -303,6 +295,21 @@ def __init__(
                 self.alpha,
             )
 
+    def _calculate_deltas(
+        self,
+        times: str | np.ndarray | FrameOrSeries | None,
+        halflife: float | TimedeltaConvertibleTypes | None,
+    ) -> None:
+        # error: Item "str" of "Union[str, ndarray, FrameOrSeries, None]" has no
+        # attribute "view"
+        # error: Item "None" of "Union[str, ndarray, FrameOrSeries, None]" has no
+        # attribute "view"
+        _times = np.asarray(
+            times.view(np.int64), dtype=np.float64  # type: ignore[union-attr]
+        )
+        _halflife = float(Timedelta(halflife).value)
+        self._deltas = np.diff(_times) / _halflife
```

Comment:
can you return _deltas and assign in the caller (change the return type as well)